theano/gof/vm.py

```diff
@@ -894,6 +935,77 @@ def make_all(self, profiler=None, input_storage=None,
             compute_map[k] = [k.owner is None]
 
         thunks = []
+
+        # Collect Reallocation Info
+        viewed_by = {}
+        for var in fgraph.variables:
+            viewed_by[var] = []
+        view_of = {}
+        reallocated_info = {}
+        dependencies = getattr(fgraph.profile, 'dependencies', {})
+
+        for idx in range(len(order)):
+            node = order[idx]
+            dmap = getattr(node.op, 'destroy_map', None)
+            vmap = getattr(node.op, 'view_map', None)
+            reallocated_ins = set([])
+
+            idx_o = 0
+            for out in node.outputs:
+                ins = None
+                if dmap and idx_o in dmap:
+                    idx_v = dmap[idx_o]
+                    assert len(
+                        idx_v) == 1, "Here we only support the possibility to destroy one input"
+                    ins = node.inputs[idx_v[0]]
+                if vmap and idx_o in vmap:
+                    assert ins is None
+                    idx_v = vmap[idx_o]
+                    assert len(
+                        idx_v) == 1, "Here we only support the possibility to view one input"
+                    ins = node.inputs[idx_v[0]]
+                if ins:
+                    assert isinstance(ins, theano.Variable)
+                    origin = view_of.get(ins, ins)
+                    view_of[out] = origin
+                    viewed_by[origin].append(out)
+                idx_o += 1
+
+            for ins in node.inputs:
+                assert not (ins in view_of and viewed_by[ins])
+                if (getattr(ins, 'ndim', None) == 0 and not storage_map[ins][0] 
+                    and ins not in fgraph.outputs and ins.owner 
+                    and dependencies.get(ins, None)):
```

Comment:
Here you are missing a condition. Maybe there is still some computation that isn't done will use ins. You  need to add back a condition that was in the file profiling.py like this:

all([compute_map[v][0] for v in dependencies[ins]])):

Here you need to create a copy of compute_map for this and you need to update it after a node was executed.